js/piwik.js

```diff
@@ -4012,6 +4060,7 @@ if (typeof Piwik !== 'object') {
              */
             detectBrowserFeatures();
             updateDomainHash();
+            setVisitorIdCookie();
```

Comment:
If this line is commented then following tests fail:

![failed test after removing line](https://cloud.githubusercontent.com/assets/466765/8026648/f8165228-0dd2-11e5-8c89-0781f835dc55.png)

These tests were written to document the existing behaviour. I'm not 100% sure that these tests are correct but I think they are mostly correct... Let me know if you want more info